lua/gitlinker.lua

```diff
@@ -22,17 +22,30 @@ local Defaults = {
   --- @type table<string, KeyMappingConfig>
   mapping = {
     ["<leader>gl"] = {
-      router = require("gitlinker.routers").blob,
       action = require("gitlinker.actions").clipboard,
       desc = "Copy git link to clipboard",
     },
     ["<leader>gL"] = {
-      router = require("gitlinker.routers").blob,
       action = require("gitlinker.actions").system,
       desc = "Open git link in browser",
     },
   },
 
+  -- different web sites use different urls, so we want to auto bind these routers
+  --
+  -- **note**:
+  -- auto bindings only work when `router=nil` in `link` API.
+  --
+  -- github.com: `/blob`
+  -- gitlab.com: `/blob`
+  -- bitbucket.org: `/src`
+  --
+  router_binding = {
+    ["^github"] = require("gitlinker.routers").blob,
+    ["^gitlab"] = require("gitlinker.routers").blob,
+    ["^bitbucket"] = require("gitlinker.routers").src,
```

Comment:
As an initial implementation this pattern match will work. Please consider the self hosted setup for gitlab and bitbucket where these patterns won't match. Consider this as a feature request, people might be interested in.

We can ask user to define the custom pattern for their host similar to this [fubitive.vim](https://github.com/tommcdo/vim-fubitive/blob/c85ca8fa2098aa05e816f5d0839a0dad6bfcca5a/plugin/fubitive.vim#L27)

Reply:
hi @G0V1NDS , I submit PR: https://github.com/linrongbin16/gitlinker.nvim/pull/118, would you please help review?